src/test/java/unit/uk/gov/ida/verifyserviceprovider/services/NonMatchingAssertionServiceTest.java

```diff
@@ -198,61 +201,23 @@ public void shouldThrowExceptionIfAssertionVersionInvalidWhenValidatingIdpAssert
 
     @Test
     public void shouldNotThrowExceptionsWhenAssertionsAreValid() {
-        List<Assertion> assertions = asList(
-                aMatchingDatasetAssertionWithSignature(emptyList(), anIdpSignature(), "requestId").buildUnencrypted(),
-                anAuthnStatementAssertion(IdaAuthnContext.LEVEL_2_AUTHN_CTX, "requestId").buildUnencrypted());
+        Assertion authnAssertion = anAuthnStatementAssertion(IdaAuthnContext.LEVEL_2_AUTHN_CTX, "requestId").buildUnencrypted();
+        Assertion mdsAssertion = aMatchingDatasetAssertionWithSignature(emptyList(), anIdpSignature(), "requestId").buildUnencrypted();
 
-        nonMatchingAssertionService.validate(assertions,"requestId", LevelOfAssurance.LEVEL_1);
+        nonMatchingAssertionService.validate(authnAssertion, mdsAssertion,"requestId", LevelOfAssurance.LEVEL_1);
 
         verify(subjectValidator, times(2)).validate(any(), any());
         verify(hubSignatureValidator, times(2)).validate(any(), any());
     }
 
-    @Test
```

Comment:
I disagree - I wrote the tests because we were passing a collection of Assertions into the translate method, and it'd be easy for one to get left out, but now the method takes two Assertions as individual arguments.  The Assertions have already been individually extracted, and then validated, by this point; there's no realistic prospect of them being null when we try and translate them.

Reply:
There is still logic in the `get` methods for the assertions though that test the extraction of the authn and mds assertion from the list. I don't think tests for validate need to be testing that logic but _something_ probably should if it is behaviour that we rely on.

Reply:
Okay.  I've now raised a tech debt card for this.